tests/components/thethingsnetwork/test_config_flow.py

```diff
@@ -0,0 +1,120 @@
+"""Define tests for the The Things Network onfig flows."""
+
+from ttn_client import TTNAuthError
+
+from homeassistant.components.thethingsnetwork.const import (
+    CONF_API_KEY,
+    CONF_APP_ID,
+    CONF_HOSTNAME,
+    DOMAIN,
+    TTN_API_HOSTNAME,
+)
+from homeassistant.config_entries import SOURCE_REAUTH, SOURCE_USER
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from . import add_schema_suggestion
+from .conftest import API_KEY, APP_ID, CONFIG_ENTRY, HOSTNAME
+
+USER_DATA = {CONF_HOSTNAME: HOSTNAME, CONF_APP_ID: APP_ID, CONF_API_KEY: API_KEY}
+USER_DATA_PARTIAL = {CONF_APP_ID: APP_ID, CONF_API_KEY: API_KEY}
+
+
+async def test_user(hass: HomeAssistant, mock_TTNClient) -> None:
+    """Test user config."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    schema = result["data_schema"]
+    user_data = schema(add_schema_suggestion(schema.schema, USER_DATA_PARTIAL))
+    assert user_data[CONF_HOSTNAME] == TTN_API_HOSTNAME  # Default value
+
+    user_data[CONF_HOSTNAME] = HOSTNAME  # Change default value
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data=user_data,
+    )
+    assert result["type"] == FlowResultType.CREATE_ENTRY
```

Comment:
```suggestion
    assert result["type"] is FlowResultType.CREATE_ENTRY
```
Please don't forget these